### PR TITLE
Fixed Antiaim on the knife and grenades

### PR DESCRIPTION
**Pull request by huoji120**
Fixed Antiaim on the knife and grenades
not fixed Defuse Bomb whit Antiaim mod.

### CODE REVIEW

**Comment by QWp6t** [comment]
lol wtf why wouldn't you just use the `IsGun()` method that already exists?

**Comment by huoji120** [comment]
@QWp6t oh , ok

File: csgo_internal/csgo_internal/AntiAim.cpp [hunk 1]
```diff
@@ -25,6 +25,16 @@ void CAntiAim::Run()
 	if( G::LocalPlayer->GetMoveType() == MOVETYPE_LADDER || G::LocalPlayer->GetMoveType() == MOVETYPE_NOCLIP )
 		return;
 
+	auto pWeapon = G::LocalPlayer->GetWeapon();
+	if (!pWeapon->GetWeaponTypewitham()) //如果返回FALSE
```
Comment (Wrapzii) [inline csgo_internal/csgo_internal/AntiAim.cpp]: use IsGun() instead of GetWeaponTypewitham() and delete GetWeaponTypewithaim()

Comment (Wrapzii) [inline csgo_internal/csgo_internal/AntiAim.cpp]: also when pulling out knife /etc you will freeze up and not able to move until AA is turned off so this is basically useless

File: csgo_internal/csgo_internal/Entity.cpp [hunk 2]
```diff
@@ -373,7 +373,68 @@ float CBaseCombatWeapon::GetWeaponSpread()
 	typedef float( __thiscall* OriginalFn )( void* );
 	return U::GetVFunc<OriginalFn>( this, 479 )( this );
 }
+bool CBaseCombatWeapon::GetWeaponTypewitham()
```
Comment (Wrapzii) [inline csgo_internal/csgo_internal/Entity.cpp]: everything on this page is a copy of whats below it.... IsGun()

**Pull request closed without merging**